packages/core/src/storage/vectorStore/ChromaVectorStore.ts

```diff
@@ -0,0 +1,133 @@
+import { VectorStore, VectorStoreQuery, VectorStoreQueryMode, VectorStoreQueryResult } from "./types";
+import { BaseNode, Document, MetadataMode } from "../../Node";
+import { ChromaClient, Collection, IncludeEnum, Where, WhereDocument, Metadata, Embeddings } from 'chromadb'
+
+type ChromaDBParams = {
+  collectionName: string
+}
+
+type ChromaDeleteOptions = {
+  where?: Where;
+  whereDocument?: WhereDocument;
+}
+
+type ChromaQueryOptions = {
+  whereDocument?: WhereDocument
+}
+
+export class ChromaVectorStore implements VectorStore {
+  storesText: boolean = true;
+  private chromaClient: ChromaClient
+  private collection: Collection | null;
+
+  constructor(params: ChromaDBParams) {
+    this.chromaClient = new ChromaClient();
+    this.collection = null;
+    this.chromaClient.createCollection({
+      name: params.collectionName,
+    }).then((collection) => {
+      this.collection = collection;
+    }).catch((err) => {
+      console.error(err);
+    }
+    );
+  }
+
+  client(): ChromaClient {
+    return this.chromaClient;
+  }
+
+  async add(embeddingResults: BaseNode[]): Promise<string[]> {
+    if (!this.collection) {
+        throw new Error("Must initialize collection before adding.");
+    }
+
+    // Check if all nodes have an embedding and only if they do, add the embeddings to the collection
+    const allNodesHaveEmbeddings = embeddingResults.every((node) => node.embedding != undefined);
```

Comment:
`add` should throw an error if any passed node doesn't have an embedding. 

Please have a look at the MongoDB implementation. It
1. Generates the data to insert 
2. Validates the generated data (calling `node.getEmbedding()` throws an error)
3. Inserts the data into the DB